File: DailyChallenges/Scala/LeetCode_1456.scala
object Solution {
    def maxVowels(s: String, k: Int): Int = {
        var ans = s.substring(0,k).count("aeiou".contains(_))
        var lc = ans;
        for (i <- k until s.length){
            if ("aeiou".contains(s(i))) lc += 1
            if ("aeiou".contains(s(i-k))) lc -= 1
            ans = lc max ans
            if (ans == k) return k;
        }
        ans
        //s.sliding(k).map(_.count("aeiou".contains(_))).max
    }
}

File: DailyChallenges/CPP/LeetCode_1456.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        string v="aeiou";
        int pre=0, res=0, cur=0;
        for(int i=0; i<s.size(); i++) {
            if(v.find(s[i])!=std::string::npos) cur++;
            if(i>=k && v.find(s[i-k])!=std::string::npos) cur--;
            res = max(res, cur);
            if (res == k) return k;            
        }
        return res;
    }
};
